### ISSUE TITLE
🗺️ A vision of Atlas

### ISSUE THREAD

**Issue by markwhiting** [Watts-Lab/atlas#1]
We argue for commensurable knowledge, and invest a substantial amount of time and effort into research cartography to that end, and yet aspects of this work are highly specialized and reify fundamental challenges to commensurability.

The knowledge atlas aims to unify our cartography efforts and to make research cartography an accessible and extensible activity for all researchers.

## Architecture
At the outset, there are 3 conceptual pieces:

- [ ] #3
- [ ] #4
- [ ] #5

The data is a common storage of all cartographic results, however an approximate abstraction of it is that it is a large table wherein there is a row for each paper (ever) and a column for each property we might use to differentiate papers. In essence, this is a system for desegregating pieces of research in a reliable way, through a systematic process of adding columns. 

Views are mechanisms that allow people to contribute new columns or view results of queries to the data. Conceptually this is similar to the `survey → csv` workflow we might have with Qualtrics or some of our other existing mapping technologies, but the view system would be designed specifically around supporting cartographic activities.

Automations are avenues for this valuable data product to get better without human intervention. As an example, products like [Elicit](https://elicit.org) add new algorithmically driven columns and we might want to sample from some of those in our mapping work. An example of an automation might be a process that extracts all a paper's references and the papers that cite it.

## Units of analysis
A critical concern in our mapping work has been navigating multiple levels of units of analysis. With this system I hope we can simplify to 3 main conceptual units, with the flexibility to extend beyond those as needed. The units I propose are driven by the possibility of defining them relatively consistently across a very large range of research activity:

- [ ] `publication`, the most common high level unit of research that we deal with
- [ ] `claim`, a single instance of expressing a potential piece of knowledge expressed in a paper, based on data, an argument or some other type of evidence or assertion.
- [ ] `observation`, a single instance of something that was measured in a research process.

GraphQL provides nice flexibility around specialized grouping factors, so we could add custom units as needed, e.g., studies and books. Studies, for example, are used differently in different fields and styles of papers. So instead of defaulting to study level analysis, leaning on a claim level analysis is likely more robust. 

Of course, a core value of providing different units of analysis is allowing for views that utilize per unit features. For example, if we wanted to document each task in a set of papers that had team tasks, we might generate a unit called `task`, that has a set of properties, such as the mapping dimensions we have ended up with in task mapping. Our view system would handle situations where a paper could have multiple tasks (or any other unit).

## Using atlas
A system like this might be used in several different ways. Whenever we read a paper we might be encouraged to add an entry into atlas with a workflow that might be as simple as entering a DOI (which can be used to look up other bibliographic information), and jotting down any notes the reader has. In the case that something about the paper emerges as interesting to a particular type of column, a process can be launched to establish the value for it. Of course in most cases, the new column would be something that needs no particular process automation, such as "the number of participants in this work". 

In a more sophisticated use, say if we were mapping a particular domain such as team tasks, we would generate a view specific to that mapping effort constructed of existing column's processes (read survey questions, but of course this could be other things), and new column processes. As we do this, we would have a validation process to establish and refine a column's procedure — something that we have done with various levels of repeatability in our attempts so far.

As a user trying to draw inference from the map, queries could call specific areas, e.g., the team task map, or could aggregate across a broader focus, e.g., to study effect sizes across all the social science papers in the atlas.

Another interesting use case is people outside our research team contributing, extending, or using the data. This seems like an extremely valuable goal for the broader community and one I intend to design around throughout this process.

## Design process
This issue aims to serve as an initial vision. One that will be refined through feedback and refined as we build parts of this and try it out. I am intending to get a simple version off the ground quickly in a combination of a gradual layer 🥞 and slice 🍰 expansion. In other words, not all features will work on day one, and not all domains will work either, but the ground work will intend to make both those areas of expansion entirely plausible.

A key design point will be an incredibly strong focus on generating repeatable processes and self documenting scientific outcomes.

A few initial questions:
- [ ] Is there something obviously wrong with this idea that we appear to be overlooking?
- [ ] #6
- [ ] #7
- [ ] How do you think this data would be most useful? For example, a lot of the mapping dimensions will only apply to a small number of papers, will there still be value in thinking about those more broadly? Is there a limit to our goal of commensurable science driven by conceptual foci?

**Comment by JamesPHoughton** [comment]
It does seem a bit like trying to create a general structure for representing arbitrary knowledge. This has been attempted, by well-financed groups - how can we keep it manageable?

**Comment by linneagandhi** [comment]
Unit breakdown: In some ways I could see claim & observation (or "unique data collection opportunity" as we've defined it) being flipped. I could see it going both ways!

**Comment by linneagandhi** [comment]
One part that would be great to solve here is what do you do as you improve the atlas thingie -- for instance, I've done 3 batches of our coding process now. Each week we have edits. Which means then I need to go back and change batch 1, batch 2, etc. which is a PAIN.

**Comment by JamesPHoughton** [comment]
In a way, the goal is to create a map over a space, and we have two major ways to think about doing that. One is to try to take the existing literature and structure it into the space, and take what lessons we get from it to tell us about the response surface in that space. Another way is to fix the space and the outcomes of interest, and then collect new data over the space. We're trying both in the different projects, and while my prior expectation was that using the literature should be less work than collecting new data over the space, now I'm not sure. At this point, I don't think I could give a conclusive prediction as to which route is easier.

**Comment by markwhiting** [comment]
> It does seem a bit like trying to create a general structure for representing arbitrary knowledge. This has been attempted, by well-financed groups - how can we keep it manageable?

Here's one example haha! https://theconversation.com/the-galactica-ai-model-was-trained-on-scientific-knowledge-but-it-spat-out-alarmingly-plausible-nonsense-195445

But more seriously, I think one thing that we are doing differently is doing it ground up (i.e., on studies we are doing), as opposed to trying to take on everything at once.

**Comment by markwhiting** [comment]
> I had been thinking that a lit review was a sortof precursor to the real work of doing new experiments. But now, I'm not thinking about it as an easier way to get a head start on an experimental paradigm, but as something that we should take on (or not) for what it can give us on its own.

Yes, I actually think theres a whole slew of long run aspects of this proposal that we haven't talked about yet, e.g., seeing new experiments as strictly contributing data to columns, and having units that represent theories as opposed to aspects of papers. This is something I don't quite want to get tied up in yet but I do think that new and exploratory experiments should be part of the conceptualization. 

Also, more generally, I think I strongly agree that the ground up vs top down aspects are tricky here. My hope is that we can start ground up and slowly think about features that can be top down (e.g., computational ones)

**Comment by markwhiting** [comment]
A note to myself that I need to expand on the values of this and on the aspect of iterative column development and column quality features.

**Comment by linneagandhi** [comment]
@markwhiting I really appreciated our convos this week and they made me even more inspired by this vision -- and I think trying it is totally worth while. I wonder what would be helpful in terms of supporting you? Would you like to set up a time before winter break where I walk you through the changes we made every week to our two processes, to give you more illustrations? Maybe we have a standing lunch or coffee together next semester as well, for faster exchange of ideas? (We could wrap in other topics too)

**Comment by markwhiting** [comment]
That would be wonderful! We can use that part of the cafe where we literally need to stand at the quasi-table.

I think walking through your workflow and changes would be great. I'm trying to convert some of the concerns that have come up into a series of design efforts (which is working but slow). As we discussed on Tuesday, the two hard parts now are the column refinement and verification workflow, and a good way of constraining units so that we don't just have an unmanageable number of custom units across different mapping efforts.

There's a lot of tricky stuff in both those problems, but I think if we can resolve them reasonably we will be much closer to making this a reality.

**Comment by linneagandhi** [comment]
Perfect! Also @markwhiting another fun update - I just got off of a call with my friend at Indeed who is really intrigued about an atlas for within-company stuff. He actually asked if we would be making a portable method or infrastructure. If you think it would be helpful to connect with him and chat ideas, he's available.

**Comment by markwhiting** [comment]
I'd love that, thanks! Perhaps after a few more iterations of thinking, e.g., in a few weeks.

**Comment by markwhiting** [comment]
Just noting Abdullah's comment that what we are really getting at with the `observation` unit above is the most granular unit within a given context (which might be very different depending on the kind of study, e.g., meta analysis vs experiment paper). 

More generally I think granularity is perhaps a useful sense for thinking about how units relate, although there are situations where things overlap across granularity levels, e.g., cluster randomized studies.

**Comment by linneagandhi** [comment]
@markwhiting flagging here: https://paperauthoringtool.com/ 
Which is a similar idea of how to get people to input data and then they get a written paper out of it (but all data is captured in a structure DB)
